### DISCUSSION TITLE
How to migrate JWT app?

### DISCUSSION THREAD

**Discussion by anhtran917** [community/community#58034] [Programming Help] [answered]
### Body

I received this email from Zoom.
**The JWT app type will be deprecated on June 1, 2023. After this date, you will no longer be able to create new JWT apps. The expected end-of-life for JWT apps will be September 1, 2023. This means that after September 1, 2023, you will no longer be allowed to interact with Zoom systems via the JWT app.
We recommend that you [create Server-to-Server OAuth or an OAuth app](https://click.zoom.us/e/84442/ternal-apps-jwt-app-migration-/c53w3x/4657961751?h=huURwscJWvJFnI3061wmUQ3AVIo5Tc1btmX4mLm4VFk) to replace the functionality of a JWT app in your account.**
I am not familiar with it and how to fix it?
Please help me.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Sandro642** [comment, marked as answer]
I understand that you have received an email from Zoom regarding the deprecation of JWT (JSON Web Token) app type and the recommendation to replace it with Server-to-Server OAuth or an OAuth app. Here's how you can address this issue:

Familiarize yourself with OAuth: OAuth is an authorization framework that allows applications to access user data on behalf of the user without sharing their credentials. Take some time to understand the concept of OAuth and how it works.

Review the Zoom documentation: Visit the Zoom developer documentation or support resources to learn more about Server-to-Server OAuth and OAuth app setup. They should provide detailed instructions and guidelines on how to create and configure these types of apps.

Assess your integration requirements: Evaluate your current use of the JWT app and determine what functionality you need to replicate using Server-to-Server OAuth or an OAuth app. Consider any specific features or permissions your integration requires and ensure that the new app type can fulfill those needs.

Create and configure a new app: Follow the instructions provided by Zoom to create a Server-to-Server OAuth or an OAuth app in your Zoom account. This process typically involves registering the app, setting up the necessary permissions, and obtaining the required credentials or access tokens.

Update your integration: Once you have the new app set up, update your integration code to use the appropriate API endpoints and authentication methods associated with Server-to-Server OAuth or OAuth. Replace any references to the deprecated JWT app with the new app's credentials and endpoints.

Test and verify: Thoroughly test your integration with the new app to ensure that it functions correctly and maintains the desired functionality. Verify that you can interact with Zoom systems as expected using the new authentication method.

Implement the changes before the deprecation date: As mentioned in the email, the expected end-of-life for JWT apps is September 1, 2023. It's important to complete the transition to Server-to-Server OAuth or an OAuth app before this date to avoid any disruptions in your integration.

If you encounter any specific challenges or require further assistance during this process, it would be best to reach out to Zoom's developer support or consult their documentation for detailed guidance tailored to your situation.

**Reply by Sandro642** [reply]
please answered this responses if is good ;)

Good luck

**Reply by anhtran917** [reply]
Thanks, Sandro

**Reply by Sandro642** [reply]
No problem ;).

**Comment by tuves** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answering and locking the post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
